game-core/src/test/resources/Total_World_War_Dec1941.xml

```diff
@@ -7749,22 +7809,32 @@
       <option name="consumesUnits" value="1:Material"/>
       <option name="requiresUnits" value="germanCombatEngineer"/>
     </attachment>
-    <attachment name="unitAttachment" attachTo="germanInfantry" javaClass="games.strategy.triplea.attachments.UnitAttachment" type="unitType">
+    <attachment foreach="Infantry:Players:Barracks" name="unitAttachment" attachTo="$Infantry$" javaClass="games.strategy.triplea.attachments.UnitAttachment" type="unitType">
```

Comment:
Instead of for-each, what if we made `attachTo` an optional nested element?

eg:
```
<attachment name="unitAttachment" javaClass="games.strategy.triplea.attachments.UnitAttachment" type="unitType">
      <option name="movement" value="1"/>
      <option name="movement" value="1"/>
      <option name="attack" value="2"/>
      <option name="attack" value="2"/>
      <option name="defense" value="3"/>
      <option name="defense" value="3"/>
      <option name="transportCost" value="2"/>
      <option name="transportCost" value="2"/>
      <option name="isInfantry" value="true"/>
      <option name="isLandTransportable" value="true"/>
      <option name="canBeGivenByTerritoryTo" value="$Players$"/>
      <option name="requiresUnits" value="$Barracks$"/>
      <option name="canInvadeOnlyFrom" value="none"/>

      <attach-to>
          <element>Infantry</element>
          <element>Players</element>
          <element>Barracks</element>
      </attach-to>
 </attachment>
```

This could potentially be a lot simpler compared to having variables, looping and placeholders.

Reply:
Not sure that makes sense. How would you know what to attach this to in your example? It would seem to indicate attaching it to all infantry, all players, and all barracks?

Reply:
For a pure XML theory I can see what you are getting at. But within how XML has been used for TripleA maps, I think its easier to understand and read for map makers to have it as an attribute.

Reply:
Are there any other examples where we use multi-valued attributes? To another extent, if we use attributes poorly in other places, I don't think it's reason to continue that practice.

Encoding multiple values in a single string and using that to do looping strikes me as very overly complex.

Reply:
Tons. If you do a quick glance through say just the UnitAttachment properties you'll see many cases that are much more complex than this in terms of multiple value attributes: https://github.com/triplea-maps/the_pact_of_steel/blob/master/map/games/pact_of_steel_2.xml#L1672

Reply:
Is that the right example @ron-murhammer ? Sorry, I think I might be missing what you're referring to. I see these options:
```
    <attachment name="unitAttachment" attachTo="infantry" javaClass="games.strategy.triplea.attachments.UnitAttachment" type="unitType">
      <option name="movement" value="1"/>
      <option name="transportCost" value="2"/>
```

That is an example though of using nested elements. Hence some small confusion on my part.

Going to a non-idiomatic form would look like this:
```
    <attachment 
       name="unitAttachment" 
      attachTo="infantry" 
      javaClass="games.strategy.triplea.attachments.UnitAttachment" 
      type="unitType"
      option="movement:1;transportCost:2" />
```
Personally I think examples like the above would do well to be unrolled to reduce complexity, hence my objection to a foreach loop attribute.

Reply:
You need to look down into some of the more complex unit options. This one and the ones following it show an example: https://github.com/triplea-maps/the_pact_of_steel/blob/master/map/games/pact_of_steel_2.xml#L1750

Reply:
Ah, thanks for the clarification. Kinda sad to see the examples, but not terribly surprised. I do think that is abuse of XML structure and attributes, but my 2 cents, I acknowledge it's not a perfectly coded world.

Reply:
Also, if you really want foreach to be a separate element instead of attribute, its probably better as a parent element of attachment not a child. So something like his:
```
<foreach>
          <element>$Infantry$</element>
          <element>$Players$</element>
          <element>$Barracks$</element>
          <attachment .../>
</foreach>
```